src/main.cpp

```diff
@@ -2930,6 +2930,22 @@ static DisconnectResult DisconnectBlock(const CBlock& block, CValidationState& s
         }
     }
 
+    // Grab the latest subtree (according to the view) and use it
+    // to determine if the block being disconnected was responsible
+    // for completing a subtree. If so, we'll pop the subtree.
+    {
+        auto disconnectSubtrees = [&] (ShieldedType type) {
```

Comment:
```suggestion
        auto disconnectSubtree = [&] (ShieldedType type) {
```

Reply:
Yeah, because we are going to add orchard support in a future PR and it will be the same logic just for a different shielded type.

Reply:
@nathan-at-least I believe the lambda is there because we *will* be invoking it twice once we add Orchard handling here when https://github.com/zcash/zcash/pull/6431 is rebased.